src/rgw/rgw_rados.cc

```diff
@@ -12843,13 +12843,13 @@ int RGWRados::cls_bucket_list(RGWBucketInfo& bucket_info, int shard_id, rgw_obj_
     *is_truncated = (*is_truncated || iter->second.is_truncated);
   }
 
-  // Create a map to track the next candidate entry from each shard, if the entry
+  // Create a vector to track the next candidate entry from each shard, if the entry
   // from a specified shard is selected/erased, the next entry from that shard will
   // be inserted for next round selection
-  map<string, size_t> candidates;
+  vector<size_t> candidates;
```

Comment:
aren't we relying on std::map to sort these candidates by key?

Reply:
@cbodley you are right, I will close pr.

I thought "m" can sort the objects by key, so "candidates" is useless. But this is wrong when "count < num_entries".